File: frontend/types/company.ts
export interface CompanyId {
  value: number;
}

export interface CompanyName {
  value: string;
}

export interface Company {
  companyId: number;  // @JsonValueのため直接数値
  companyName: string;  // @JsonValueのため直接文字列
  withholdingTax: boolean;
  freeePartnerId: number | null;
}

export interface CompanyListResponse {
  companyList: Company[];
}

File: backend/src/main/java/com/kamiokaweb/redo/model/company/Company.java
package com.kamiokaweb.redo.model.company;

import io.swagger.v3.oas.annotations.media.Schema;

@Schema(description = "会社")
public record Company(
        CompanyId companyId,
        CompanyName companyName,
        @Schema(description = "源泉徴収有無") Boolean withholdingTax
) {
}
